Assets/Scripts/Gameplay/Action/BaseTypes/Action.cs

```diff
@@ -24,54 +30,97 @@ namespace Unity.Multiplayer.Samples.BossRoom.Actions
     ///
     /// Note also that if Start() returns false, no other functions are called on the Action, not even End().
     /// </remarks>
-    public abstract class Action : ActionBase
+    public abstract class Action : ScriptableObject
     {
-        protected ServerCharacter m_Parent;
+        /// <summary>
+        /// A reference to the prototype action that helps us identify actions of the same type
+        /// </summary>
+        [NonSerialized]
+        public Action RuntimePrototypeReference;
```

Comment:
Could be worth adding some more comments here on the lifecycle of an action now and how it works with that reference. Just to understand the relationship between the prototypeRef, the scriptable object, the runtime instances, etc.

Reply:
Turned out to be an unneeded field, so I'm removing it. Will add said comment above the ActionID field.